src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -75,8 +75,11 @@ public ProjectRestoreCommand(ILogger logger, ProjectRestoreRequest request)
                 _request.MaxDegreeOfConcurrency,
                 token);
 
+            // The user global package folder is always ordered first.
+            var userPackageFolder = localRepositories.First();
```

Comment:
I've updated this, it will flow the userFolder/fallbacks separately down from the restore request.